sympy/core/mul.py

```diff
@@ -1249,25 +1249,27 @@ def _eval_is_zero(self):
 
     def _eval_is_integer(self):
         from sympy import fraction
+        from sympy.core.numbers import Float
 
         is_rational = self._eval_is_rational()
         if is_rational is False:
             return False
 
         n, d = fraction(self)
-        if not is_rational:
-            _self = n/d
-            if _self != self:
-                return _self.is_integer
         if is_rational:
             if d is S.One:
                 return True
-            elif d == S(2):
+        if d.is_even:
+            if d.is_prime:
```

Comment:
maybe we should try be quicker:
`if d.is_Integer and d == 2:`
It's a very small corner case to create a symbolic 2 as an even prime. I still don't like allowing symbolic  singletons -- if assumptions allow the identity to be determined it seems like that value -- not symbol -- should be substituted for the Symbol at instantiation. @oscarbenjamin, isn't this a dilemma? Make everything slower to test for symbolic singletons or else don't allow symbolic singletons.

Reply:
```
>>> symbols('Two', even=True,prime=True).equals(2)
False
```
a tradeoff for symbolic support ? 😁

Reply:
if ☝️ is true then `d == S(2)` would be faster and have symbolic support ?

Reply:
> ```
> >>> symbols('Two', even=True,prime=True).equals(2)
> False
> ```
> 
> a tradeoff for symbolic support ? 😁

Cause there's only one even prime why doesn't sympy support this ?